app/controllers/concerns/liquid_enabled_resource.rb

```diff
@@ -0,0 +1,39 @@
+module LiquidEnabledResource
+  extend ActiveSupport::Concern
+
+  included do
+    helper_method :liquid_assigns
+  end
+
+  def liquid_assigns
+    @liquid_assigns ||= default_liquid_assigns.merge!(liquid_resource_assigns)
+  end
+
+  # To be overwritten by each controller
+  def liquid_resource_assigns
+    raise 'Controller included LiquidEnabledResource without defining a liquid_resource_assigns'
+  end
+
+  private
+
+  def default_liquid_assigns
+    if params[:project_id]
+      project_assigns
+    else
+      {}
+    end
+  end
+
+  def project_assigns
+    result = {}
+
+    # This is required because we may be in Markup#preview that's passing
+    # :project_id for Tilyum rendered editors
+    project = Project.find(params[:project_id])
+    # authorize! :use, project
```

Comment:
I'm assuming this is a Pro-only code. If so, can we go with this for parity:

```suggestion
    if defined?(Dradis::Pro)
      authorize! :use, project
    end
```

Reply:
I dislike those types of checks, but it's a good point re parity, I wonder what would it take (if anything) to have the `authorize!` call just work in CE.

Reply:
>  wonder what would it take (if anything) to have the authorize! call just work in CE.

The Pro CE ability files are already different by a large margin. We could likely add a `can :manage, Project` in CE which would solve this?

Reply:
Resolved in [a new PR](https://github.com/dradis/dradis-ce/pull/873/).